moveit_experimental/moveit_jog_arm/include/moveit_jog_arm/jog_calcs.h

```diff
@@ -128,6 +128,15 @@ class JogCalcs
 
   void insertRedundantPointsIntoTrajectory(trajectory_msgs::JointTrajectory& trajectory, int count) const;
 
+  /**
+   * Remove the Jacobian row and the delta-x element of one Cartesian dimension, to take advantage of task redundancy
+   *
+   * @param matrix The Jacobian matrix.
+   * @param delta_x Vector of Cartesian delta commands, should be 6-long.
+   * @param row_to_remove Dimension that will be allowed to drift, e.g. row_to_remove = 2 allows z-translation drift.
+   */
+  void removeDimension(Eigen::MatrixXd& matrix, Eigen::VectorXd& delta_x, unsigned int row_to_remove);
```

Comment:
Seems like this should be two different methods:
```suggestion
  void removeDimension(Eigen::MatrixXd& matrix, unsigned int row_to_remove);
  void removeDimension(Eigen::VectorXd& delta_x, unsigned int index_to_remove);
```

Reply:
I'm not a fan of 2 separate methods because:

- you need to call both methods or there will be a size mismatch when you try to multiply them

- it's a little bit more efficient this way  (same for-loop)